### PR TITLE
Issue 27 fix: findSequencesOnDisk can display hidden files as an option, not the default

### PR DESCRIPTION
**Pull request by michaelmovies**
This re-asserts the old behavior of findSequencesOnDisk regarding hidden files (which became exposed, beneficially, by recent changes to yield_sequences_in_list), and will NOT return them in the search results by default.  A kwarg is added that allows the user to get hidden files if desired.

Note: this does NOT pre-filter out files that have no extension ... these were being excluded by findSequencesOnDisk (and the underlying mechanism) previously, but that was a bug.  So, previously, if you did a search in a directory that contained only the files README and README.rst, you'd get back README.rst but not README in the results of findSequencesOnDisk ... now you will get back both.

In other words the behavior of findSequencesOnDisk has still changed (for the better), but with the default behavior remaining reasonably consistent with the actual intent.

### CODE REVIEW

**Comment by justinfx** [comment]
This is just a nitpick, but could we shorten that kwarg just a little bit? Maybe "include_hidden"? 
The function doesn't really "show" anything anyways and we make the kwarg slightly smaller.

**Comment by michaelmovies** [comment]
Updated per nitpick.
